packages/contracts-bedrock/contracts/dispute/BondManager.sol

```diff
@@ -0,0 +1,178 @@
+     */
+    event BondSeized(bytes32 bondId, address owner, address seizer, uint256 amount);
+
+    /**
+     * @notice BondReclaimed is emitted when a bond is reclaimed by the owner.
+     * @param bondId is the id of the bond.
+     * @param claiment is the address that reclaimed the bond.
+     * @param amount is the amount of the bond.
+     */
+    event BondReclaimed(bytes32 bondId, address claiment, uint256 amount);
+
+    /**
+     * @notice The permissioned dispute game factory.
+     * @dev Used to verify the status of bonds.
+     */
+    IDisputeGameFactory public immutable DISPUTE_GAME_FACTORY;
+
+    /**
+     * @notice Instantiates the bond maanger with the registered dispute game factory.
+     * @param _dgf is the dispute game factory.
+     */
+    constructor(IDisputeGameFactory _dgf) {
+        DISPUTE_GAME_FACTORY = _dgf;
+    }
+
+    /**
+     * @notice Post a bond with a given id and owner.
+     * @dev This function will revert if the provided bondId is already in use.
+     * @param bondId is the id of the bond.
+     * @param owner is the address that owns the bond.
+     * @param minClaimHold is the minimum amount of time the owner
+     *        must wait before reclaiming their bond.
+     */
+    function post(
+        bytes32 bondId,
+        address owner,
+        uint256 minClaimHold
+    ) external payable {
+        require(bonds[bondId].owner == address(0), "BondManager: BondId already posted.");
+        require(owner != address(0), "BondManager: Owner cannot be the zero address.");
+        require(msg.value > 0, "BondManager: Value must be non-zero.");
+
+        uint256 expiration = minClaimHold + block.timestamp;
+        bonds[bondId] = Bond({
+            owner: owner,
+            expiration: expiration,
+            bondId: bondId,
+            amount: msg.value
+        });
+
+        emit BondPosted(bondId, owner, expiration, msg.value);
+    }
+
+    /**
+     * @notice Seizes the bond with the given id.
+     * @dev This function will revert if there is no bond at the given id.
+     * @param bondId is the id of the bond.
+     */
+    function seize(bytes32 bondId) external {
+        Bond memory b = bonds[bondId];
+        require(b.owner != address(0), "BondManager: The bond does not exist.");
+        require(b.expiration >= block.timestamp, "BondManager: Bond expired.");
+
+        IDisputeGame caller = IDisputeGame(msg.sender);
+        IDisputeGame game = DISPUTE_GAME_FACTORY.games(
+            GameType.ATTESTATION,
+            caller.rootClaim(),
+            caller.extraData()
+        );
+        require(msg.sender == address(game), "BondManager: Unauthorized seizure.");
+        require(game.status() == GameStatus.CHALLENGER_WINS, "BondManager: Game incomplete.");
+
+        delete bonds[bondId];
+
+        emit BondSeized(bondId, b.owner, msg.sender, b.amount);
+
+        bool success = SafeCall.call(payable(msg.sender), gasleft(), b.amount, bytes(""));
```

Comment:
Should we add an overload of `call` that doesn't accept calldata?

Reply:
Yes definitely, great idea!

Reply:
`SafeCall.send`?

Reply:
Done in #5675, will refactor ontop of that :)